cognite/extractorutils/unstable/core/base.py

```diff
@@ -40,29 +46,38 @@ def __init__(
         self._checkin_lock = RLock()
         self._runtime_messages: Optional[Queue[RuntimeMessage]] = None
 
+        self._scheduler = TaskScheduler(self.cancellation_token.create_child_token())
+
+        self._tasks: list[Task] = []
+        self._task_updates: list[TaskUpdate] = []
+
         self.logger = logging.getLogger(f"{self.EXTERNAL_ID}.main")
 
     def _set_runtime_message_queue(self, queue: Queue) -> None:
         self._runtime_messages = queue
 
-    def _run_checkin(self) -> None:
-        def checkin() -> None:
-            body = {"externalId": self.connection_config.extraction_pipeline}
+    def _checkin(self) -> None:
+        with self._checkin_lock:
```

Comment:
Okay, a bit nitpicky, but I think we have to be careful about this stuff here:

If odin is not responding, this will block the entire extractor while retrying. What you _could_ do instead, is grab the checkin lock, empty the update list into a temporary, then release the lock. That way, the extractor will continue accumulating task updates while we retry against CDF.